main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1088,98 +1105,107 @@ object Typer {
           (constrs2, tpe2, eff2) <- visitExp(exp2)
           (constrs3, regionType, eff3) <- visitExp(exp3)
           _ <- unifyTypeM(regionType, Type.mkRegion(regionVar, loc), loc)
```

Comment:
Should be `exp3.loc`? I'm not totally sure what each `expN` corresponds to here.